### PR TITLE
[s] Improve Players' Impression of Coder Effeciency

### PR DESCRIPTION
**Pull request by YotaXP**
Outputs randomly generated PR announcements.

Example
- Outputs 1 PR every 2 minutes on average. <sup>(Can be tweaked by VVing the `wait` var.)</sup>
- Ignores players' "Show Pull Requests" preference. <sup>(Can be changed by VVing the `ignore_player_pref` var.)</sup>

**If you don't want your name to show up in these, speak now, or forever hold your peace!**

### CODE REVIEW

File: code/controllers/subsystem/pullrequests.dm [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		if(prob(10))
+			title += " and "
+			continue
+		else
+			break
+
+	// Capitalize first verb.
+	title[2] = "[uppertext(copytext(title[2], 1, 2))][copytext(title[2], 2)]"
+
+	if(!isMerge && prob(20))
```
Comment (ChangelingRain) [inline code/controllers/subsystem/pullrequests.dm]: if it's being merged these should still have a lower chance to happen

**Comment by Steelpoint** [comment]
>hornygranny - Addresses memes

Surprisingly accurate.

**Comment by ChangelingRain** [comment]
>i'm not in the testing picture at ALL

this picture is biased I want a new one

File: code/controllers/subsystem/pullrequests.dm [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		21;"swankcookie", 19;"lordpidey", 19;"Lobachevskiy", 19;"SuperSayu", 19;"Delimusca",
+		19;"Delimusca", 19;"spasticVerbalizer", 19;"dumpdavidson", 18;"Fox-McCloud", 18;"ACCount12",
+		18;"ACCount12", 18;"Deantwo", 18;"adrix89", 17;"octareenroon91", 17;"francinum",
+		17;"francinum", 16;"Nienhaus", 15;"CosmicScientist", 15;"as334", 15;"Neerti",
+		15;"Neerti", 14;"Strathcona", 14;"Zelacks", 13;"RandomMarine", 13;"Fayrik",
+		13;"Fayrik", 12;"Jalleo", 12;"Tokiko1", 12;"freerealestate", 12;"CollenN",
+		12;"CollenN", 11;"Cuboos", 11;"feemjmeem", 10;"Niknakflak", 10;"Alek2ander",
+		10;"Alek2ander", 10;"Cluwnes")
+
+	var/maintainer = !isMerge ? author : pick(
```
Comment (ChangelingRain) [inline code/controllers/subsystem/pullrequests.dm]: small chance of selfmerging pls

File: code/controllers/subsystem/pullrequests.dm [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		24;"Robustin", 24;"Boggart", 24;"Donkie", 24;"Limeliz", 23;"Kelenius",
+		23;"Kelenius", 22;"bear1ake", 22;"Thunder12345", 22;"caelaislinn", 22;"Mloc",
+		22;"Mloc", 22;"bgare89", 21;"Bawhoppen", 21;"YotaXP", 21;"swankcookie",
+		21;"swankcookie", 19;"lordpidey", 19;"Lobachevskiy", 19;"SuperSayu", 19;"Delimusca",
+		19;"Delimusca", 19;"spasticVerbalizer", 19;"dumpdavidson", 18;"Fox-McCloud", 18;"ACCount12",
+		18;"ACCount12", 18;"Deantwo", 18;"adrix89", 17;"octareenroon91", 17;"francinum",
+		17;"francinum", 16;"Nienhaus", 15;"CosmicScientist", 15;"as334", 15;"Neerti",
+		15;"Neerti", 14;"Strathcona", 14;"Zelacks", 13;"RandomMarine", 13;"Fayrik",
+		13;"Fayrik", 12;"Jalleo", 12;"Tokiko1", 12;"freerealestate", 12;"CollenN",
+		12;"CollenN", 11;"Cuboos", 11;"feemjmeem", 10;"Niknakflak", 10;"Alek2ander",
```
Comment (ChangelingRain) [inline code/controllers/subsystem/pullrequests.dm]: CollenN is listed twice

File: code/controllers/subsystem/pullrequests.dm [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	var/nextPr
+	var/ignore_player_pref = 1
+
+/datum/subsystem/pullrequests/New()
+	nextPr = rand(17000, 17500)
+	NEW_SS_GLOBAL(SSpullrequests)
+
+/datum/subsystem/pullrequests/fire()
+	if(prob(10))
+		announcePr()
```
Comment (MrStonedOne) [inline code/controllers/subsystem/pullrequests.dm]: Should make this a var too.

So i can vv wait to 5 and prob to 100 to spam them =P

File: code/controllers/subsystem/pullrequests.dm (hunk 3 above)
Comment (ChangelingRain) [inline code/controllers/subsystem/pullrequests.dm]: actually ones on line endings are all listed twice

**Comment by PKPenguin321** [comment]
finally, a player reference for me!

File: code/controllers/subsystem/pullrequests.dm [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+		var/list/punchlines = list("water", "organs", "polyacid", "memes", "taser",
+		                           "stamina damage", "hardsuits", "exploit", "nuke",
+		                           "ponies", "armor", "health", "uplink", "cat",
+		                           "talismans", "metagame", "hats", "run speed")
+
+		if(prob(25))
+			title += pick("the [pick("Captain's", "clown's", "HoS's", "janitor's", "changeling's", "AI's", "bloody")] ",
+			              "[pick("Pun Pun's", "Poly's", "Goon's", "goof's", "Ian's")] ")
+		else
+			punchlines += list("Boxstation", "Metastation", "iamgoofball", "AI", "fire",
```
Comment (ChangelingRain) [inline code/controllers/subsystem/pullrequests.dm]: add Dreamstation, Asteroidstation, Ministation and Birdboat to this list

**Comment by Incoming5643** [comment]
the fact that the odds of your name being attached to the pr is relative to the number of prs you've made to date is a nice touch.

Especially since I'm ranked so highly in that specific metric :smile:

File: code/controllers/subsystem/pullrequests.dm (hunk 3 above)
Comment (YotaXP) [inline code/controllers/subsystem/pullrequests.dm]: Oh heck.  My Excel wizardry failed me!

File: code/controllers/subsystem/pullrequests.dm (hunk 2 above)
Comment (YotaXP) [inline code/controllers/subsystem/pullrequests.dm]: It's already possible for the two pick()s to return the same name.

Comment (ChangelingRain) [inline code/controllers/subsystem/pullrequests.dm]: oh wait you're right

**Comment by duncathan** [comment]
That's amazing

**Comment by optimumtact** [comment]
Use the strings system that the ion laws use

**Comment by Nienhaus** [comment]
Hey cool I made the list.

**Comment by Core0verload** [comment]
@optimumtact, why bother? It's an April Fools only thing.

**Comment by YotaXP** [comment]
>Use the strings system that the ion laws use

I was not aware of this new JSON stuff.  Can I get away with it this once, since it's never going to be seen again after the 1st? :pray: 

I tried to pull the contributor data from the Github API dynamically, but it turns out `world.Export()` cannot into HTTPS.

**Comment by optimumtact** [comment]
I forgive you

**Comment by duncathan** [comment]
I love this one

**Pull request merged**